src/posix.mak

```diff
@@ -494,8 +497,11 @@ install: all
 
 ######################################################
 
-checkwhitespace: $(HOST_DMD_PATH)
-	CC=$(HOST_CXX) $(HOST_DMD_RUN) -run checkwhitespace.d $(SRC) $(GLUE_SRC) $(ROOT_SRCS)
+checkwhitespace: $(HOST_DMD_PATH) $(TOOLS_DIR)
+	CC=$(HOST_CXX) $(HOST_DMD_RUN) -run $(TOOLS_DIR)/checkwhitespace.d $(SRC) $(GLUE_SRC) $(ROOT_SRCS)
+
+$(TOOLS_DIR):
```

Comment:
This is what [order-only prerequisites](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html) are made for. I suggest using them (all dependencies after tge pipe `|` will only trigger this target if it doesn't exist)

@andralex a similar pattern is used at dlang.org:

https://github.com/dlang/dlang.org/blob/master/posix.mak#L359